libevm/VMCalls.cpp

```diff
@@ -139,12 +142,20 @@ void VM::caseCreate()
 void VM::caseCall()
 {
 	m_bounce = &VM::interpretCases;
+
+	// TODO: Please check if that does not actually increases the stack size.
+	//       That was the case before.
 	unique_ptr<CallParameters> callParams(new CallParameters());
+
+	// Clear the return data buffer. This will not free the memory.
+	m_returnData.clear();
+
 	bytesRef output;
 	if (caseCallSetup(callParams.get(), output))
 	{
 		std::pair<bool, owning_bytes_ref> callResult = m_ext->call(*callParams);
 		callResult.second.copyTo(output);
+		m_returnData.assign(callResult.second.begin(), callResult.second.end());
```

Comment:
Could we define `m_returnData` as `owning_bytes_ref` and then move here instead of copying?

(if not, using helpers like `owning_bytes_ref::toBytes` probably would make this expression simpler)